File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
import functools as ft

def find_armstrong_numbers(numbers):
    armstrong_num = []

    # Checks if number in list is an armstrong number. If so, appends to armstrong_num list
    for num in numbers:
        if(test_for_armstrong(num) == True):
            armstrong_num.append(num)

    return armstrong_num

def test_for_armstrong(numbers):
    num_ls = [int(num) for num in str(numbers)] # separate each digit into a list
    num_raised = [num**len(num_ls) for num in num_ls]    # raise each num to the power of the amount of digits in the original number
    sum = (ft.reduce(lambda agg, num : agg + num, num_raised))   # use reduce() to aggregate all nums in the armstrong_num list
    if(sum == numbers):
        return True
    else:
        return False

# print(find_armstrong_numbers(list(range(0,20000))))

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

exports.findArmstrongNumbers = function(numbers) {
    let armstrongNumbers = [];

    for(let i=0; i<numbers.length; i++) {
        let digits = String(i).split('');
        let sum = 0;
        for(let i=0; i<digits.length; i++) {
            sum += Math.pow(digits[i], digits.length)
        }

        if(sum == numbers[i]) {
            armstrongNumbers.push(numbers[i]);
        }
    }

    return armstrongNumbers;
};